theano/gpuarray/elemwise.py

```diff
@@ -1744,7 +1743,7 @@ def gpu_kernels(self, node, nodename):
             reduce_fct = self._assign_reduce(node, nodename, "myresult",
                                              load_in + "(A[i0])",
                                              {}, True)
-            reduce_init = self._assign_init(load_in + "(A[0])")
+            reduce_init = self._assign_init(load_in + "(A[0])", acc_dtype)
```

Comment:
I think you should use the input dtype, not the acc_dtype in all call to assign_init()

Reply:
OK